msproteomicstoolslib/data_structures/modifications.py

```diff
@@ -125,9 +125,9 @@ def translateModificationsFromSequence(self, sequence, code, aaLib = None) :
         if code == 'unimod' :
             aminoacides_with_mods     = re.findall('([A-Z]\([^\)]*\)|[A-Z])', sequence )
         else :
-            aminoacides_with_mods     = re.findall('([A-Z]\[\d*\]|[A-Z])', sequence )
+            aminoacides_with_mods     = re.findall('([A-Z]\[[^\]]*\]|[A-Z])', sequence )
             #Warning : this will only work for TPP unfortunately. It is though the most common operation we'll do
-            if code == 'TPP' : terminal_mods = re.findall('([a-z]\[\d*\]|[a-z])', sequence) 
+            if code == 'TPP' : terminal_mods = re.findall('([a-z]\[[^\]]*\]|[a-z])', sequence) 
```

Comment:
ok, so this now matches anything inside square brackets, not just numbers. I think that is more general and portable and also reflects the fact that people put in `[+81]` and stuff